src/Common/src/Interop/Windows/kernel32/Interop.GetFileAttributesEx.cs

```diff
@@ -68,10 +67,8 @@ internal unsafe struct WIN32_FIND_DATA
             internal uint nFileSizeLow;
             internal uint dwReserved0;
             internal uint dwReserved1;
-            [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 260)]
-            internal string cFileName;
-            [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 14)]
-            internal string cAlternateFileName;
+            internal FixedString.Size260 cFileName;
+            internal FixedString.Size14 cAlternateFileName;
```

Comment:
> you additionally need a way to know the actual size

You've defined the size in a const.  You just pass in that const.  And you look for a null terminator exactly how you're currently doing.

> It's error-prone to access fixed arrays

I think it's way more error prone to have this custom struct thing that's very non-standard and difficult for someone who stumbles upon it in the code to understand how it all works.  Anyone versed in C# can understand how fixed-sized buffers work.

> Extension methods aren't workable

They don't need to be extension methods.  They'd just be normal static helpers.